query/src/main/java/org/infinispan/query/clustered/ClusteredCacheQueryImpl.java

```diff
@@ -123,11 +123,12 @@ private int getNodeMaxResults() {
       List<QueryResponse> responses = invoker.broadcast(command);
 
       for (QueryResponse queryResponse : responses) {
-         ClusteredTopDocs topDocs = new ClusteredTopDocs(queryResponse.getTopDocs(), queryResponse.getNodeUUID());
-
-         resultSize += queryResponse.getResultSize();
-         topDocs.setNodeAddress(queryResponse.getAddress());
-         topDocsResponses.put(queryResponse.getNodeUUID(), topDocs);
+         if (queryResponse.getResultSize() > 0) {
+            ClusteredTopDocs topDocs = new ClusteredTopDocs(queryResponse.getTopDocs(), queryResponse.getNodeUUID());
```

Comment:
I understand this check is done because the queryResponse.getTopDocs() can be null. 
So when creating a ClusteredTopDocs, we can have a NullPointerException if this check is not done in the DistributedIterator.
What do you think about moving creation code into the QueryResponse so we can throw an assertionError on creation ?

In QueryReponse

``` java
public ClusteredTopDocs toClusteredTopDocs() {
      if(nodeTopDocs == null) throw new IllegalStateException("ClusteredTopDocs can't be created from an empty QueryResponse");
      ClusteredTopDocs clusteredTopDocs = new ClusteredTopDocs(this.getTopDocs(), this.getNodeUUID());
      clusteredTopDocs.setNodeAddress(this.getAddress());
      return clusteredTopDocs;
   }
```